docs/guides/implementation_blueprint/data-synchronization.md

```diff
@@ -0,0 +1,200 @@
+
+<!-- Trying this approach for a test. If it's liked, we can add it throughout the docs. Otherwise, I'll delete it. -->
+
+## Customer synchronization
+
+Voucherify can store and act on collected customer data. The most common customer data operation is to create customer groups that can be used to understand customer behavior, build promotion limits, or trigger automatic incentive delivery.
+
+<!-- ^ the WHY of importance -->
+
+Voucherify uses the [customer object](ref:customer-object) for [validation](ref:validation-object), [redemption](ref:redemption-object), and [distribution](https://support.voucherify.io/article/19-how-does-the-distribution-manager-work "How does the distribution manager work?") purposes.
+
+<!-- instead of distribution - publication? -->
+
+The customer data does not have to be stored in Voucherify before a validation or redemption request is made. Voucherify can use the customer data from the request to create a new customer with a unique `id` field.
+
+However, the customer data must exist in Voucherify before a distribution is made.
+
+<!-- distribution as meant by distribution channels, not publication, as in the postman collection? -->
+
+<!-- this bit must be moved someplace
+
+Every customer has unique identification data in the `source_id` field. The `source_id` can be a customer ID or email from a CRM system, database, or a third-party service.  It can be imported into Voucherify from your [CRM database]() or a CSV file <!-- link when ready! -->
+
+> 🚧 Customer data synchronization and `source_id`
+>
+> If your data is to be synchronized based on the `source_id` of the customer, you need to import the `source_id` when the customer is uploaded for the first time. You will not be able to change or update `source_id` later on.
+
+### Customer synchronization options
+
+<!-- https://success.voucherify.io/article/430-crm-playbook
+Adding customers
+
+These are the ways you can add customer profiles to Voucherify:
+
+API
+Webhooks
+The import with a CSV file. 
+You can integrate your CRM platform with Voucherify.
+
+ -->
+
+#### Upserting customers
+
+<!-- do we clarify what `upsert` means: `update` + `insert`? -->
+
+If the customer does not exist in Voucherify, they are created automatically in a request that includes the customer object. If the customer exists, their data will be upserted. In this case, the response also includes the `updated_at` object. <!-- these are objects, right? -->
+
+The following actions <!-- or activities? --> upsert customer data:
+- order creation
+- voucher publication
+- validation <!-- IMO should be here-->
+- redemption
+
+<!-- Redemption... but this is to be deprecated, right? so should we actually link/mention those
+
+I need clarification – we want customers to use qualification endpoint or the stackable discounts endpoint, and these do redeem, so is this info `redemption` all right?
+
+Anything else that MUST be here? I don't mean everything
+
+-->
+
+<!-- ##### Order creation
```

Comment:
At first, I thought it would need more in-depth approach, but then I changed my mind. Nevertheless, I've left it there for future use in case of feedback that these needs elaboration. Since you find the section fine, I'm going to remove those 🙂